modules/examples/codegen/protocol/src/main/scala/example/codegen/ExampleService.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package example.codegen
+
+import higherkindness.mu.rpc.protocol._
+
+@message case class Hello(words: String)
+
+@serviceCodeGen(Protobuf) trait ExampleService[F[_]] {
```

Comment:
I wonder how the code generation would behave when we would be generating code in two different stages:

1) From IDL to Scala code (scala files with `@serviceCodeGen`) (**sbt-mu-idl-gen**)
2) `@serviceCodeGen` to the expanded version of the Scala Code (**scala compiler plugin**)

Reply:
I haven't tried it yet but I suppose the SBT plugin generates the `@service`-annotated sources from IDL, and then the compiler plugin generates the service infrastructure from those sources afterwards. 

We could explore combining these two steps into one, however I don't think that would work since the developer needs the intermediate `@service`-annotated traits in order to implement their methods.